### DISCUSSION TITLE
Does java instrumentation agent follow latest semantic-conventions?

### DISCUSSION THREAD

**Discussion by missTam** [open-telemetry/opentelemetry-java-instrumentation#12529] [Q&A] [answered]
With the semantic-conventions [release v1.26.0](https://github.com/open-telemetry/semantic-conventions/releases/tag/v1.26.0) `url.template` information was added on the HTTP client metric ([PR](https://github.com/open-telemetry/semantic-conventions/pull/675)). 

Does java instrumentation agent (latest) support this semantic-convention?

How does one generally check compatibility between java-instrumentation and semantic-convention versions?

Thanks!

**Comment by laurit** [comment, marked as answer]
No, java agent does not support `url.template`. Note that vast majority of java http clients have no concept of url templates so they'll not fill this attribute. This attribute could for example be used with spring WebClient if it is used in a certain way, see https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10506 for an attempt to implement this that predates `url.template`.

> How does one generally check compatibility between java-instrumentation and semantic-convention versions?

I don't think there is an easy way. You'll have to examine the source code for definitive answers.

**Reply by missTam** [reply]
Thanks for your response!

If I follow correctly, for this attribute to become available, Spring WebClient would need to provide something like a `uriTemplate`, so you could use it in your implementation, as proposed in #10506 ?

I thought the uri-templating logic could be owned by you..?

**Reply by laurit** [reply]
For http clients that don't provide a template an extension point could be implemented where users could provide custom logic to deduce the template from an url. It may also be possible to implement this in a way that could use rules from a configuration file. Something like if url matches some regex then return some template.

**Reply by missTam** [reply]
I agree that relying on http clients to provide a template might be unrealistic to expect..

Do you envision this extension as part of the opentelemetry-java-instrumentation?

Another idea: would it make sense to say: 'We will export `url.template` for you if you provide it and it matches some regex, otherwise, we will ignore it. How you get it (Spring WebClient offers it or you extended the WebClient with your own [custom attribute](https://docs.spring.io/spring-framework/reference/web/webflux-webclient/client-attributes.html)), we don't care' ?

**Reply by laurit** [reply]
> I agree that relying on http clients to provide a template might be unrealistic to expect..

most http clients are not designed for this so they'll never provide it

> Do you envision this extension as part of the opentelemetry-java-instrumentation?

yes, but it is just an idea. It is possible that it will never get implemented.

 > Another idea: would it make sense to say: 'We will export url.template for you if you provide it and it matches some regex, otherwise, we will ignore it. How you get it (Spring WebClient offers it or you extended the WebClient with your own [custom attribute](https://docs.spring.io/spring-framework/reference/web/webflux-webclient/client-attributes.html)), we don't care' ?

I don't follow.

**Reply by trask** [reply]
one idea that has been discussed is that if/when something like https://github.com/open-telemetry/oteps/pull/207 happens, then users could set their own `url.template` before calling the HttpClient and the instrumentation would automatically pick that up and stamp it on to the telemetry
